### PR TITLE
Add python-gssapi

### PR DESCRIPTION
**Pull request by dhirschfeld**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/python-gssapi/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  build:
+    - {{ compiler('c') }}
+  host:
+    - python
+    - cython
+    - krb5
+    - pip
+  run:
+    - python
+    - decorator
```
Comment (dhirschfeld) [inline recipes/python-gssapi/meta.yaml:27]: https://github.com/pythongssapi/python-gssapi/blob/dac08a83ebe14c2b4919565628f2a7bef4ebbdd7/setup.py#L289-L291

**Comment by dhirschfeld** [comment]
```
    Exception: Could not find main GSSAPI shared library.  Please try setting GSSAPI_MAIN_LIB yourself or setting ENABLE_SUPPORT_DETECTION to 'false'
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.

```

**Comment by dhirschfeld** [comment]
OSX failed. I suspect it's down to not finding the right `gssapi` library due to it being deployed with the `conda-forge` `krb5` package rather than in a usual system location.
* https://github.com/pythongssapi/python-gssapi/blob/dac08a83ebe14c2b4919565628f2a7bef4ebbdd7/setup.py#L74-L75
* https://github.com/pythongssapi/python-gssapi/blob/dac08a83ebe14c2b4919565628f2a7bef4ebbdd7/setup.py#L90-L91

Without a mac to test I'll just exclude that platform for now and leave that as exercise for anyone interested...

<details>

```

  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/ext_ggf.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!
   INFO (python-gssapi,lib/python3.8/site-packages/gssapi/raw/ext_ggf.cpython-38-darwin.so): Needed DSO /usr/lib/libSystem.B.dylib found in the whitelist
   INFO (python-gssapi,lib/python3.8/site-packages/gssapi/raw/chan_bindings.cpython-38-darwin.so): Needed DSO /usr/lib/libSystem.B.dylib found in the whitelist
  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/message.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!
   INFO (python-gssapi,lib/python3.8/site-packages/gssapi/raw/message.cpython-38-darwin.so): Needed DSO /usr/lib/libSystem.B.dylib found in the whitelist
  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/ext_rfc5587.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!
   INFO (python-gssapi,lib/python3.8/site-packages/gssapi/raw/ext_rfc5587.cpython-38-darwin.so): Needed DSO /usr/lib/libSystem.B.dylib found in the whitelist
  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/ext_cred_imp_exp.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!
   INFO (python-gssapi,lib/python3.8/site-packages/gssapi/raw/ext_cred_imp_exp.cpython-38-darwin.so): Needed DSO /usr/lib/libSystem.B.dylib found in the whitelist
  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/misc.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!
   INFO (python-gssapi,lib/python3.8/site-packages/gssapi/raw/misc.cpython-38-darwin.so): Needed DSO /usr/lib/libSystem.B.dylib found in the whitelist
WARNING (python-gssapi): run-exports library package conda-forge::python-3.8.5-hfc71d35_5_cpython in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
Traceback (most recent call last):
  File ".ci_support/build_all.py", line 142, in <module>
    build_all(os.path.join(root_dir, "recipes"), args.arch)
  File ".ci_support/build_all.py", line 66, in build_all
    build_folders(recipes_dir, new_comp_folders, arch, channel_urls)
  File ".ci_support/build_all.py", line 123, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "/Users/runner/miniforge3/lib/python3.7/site-packages/conda_build/api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/Users/runner/miniforge3/lib/python3.7/site-packages/conda_build/build.py", line 2864, in build_tree
    notest=notest,
  File "/Users/runner/miniforge3/lib/python3.7/site-packages/conda_build/build.py", line 2151, in build
    newly_built_packages = bundlers[pkg_type](output_d, m, env, stats)
  File "/Users/runner/miniforge3/lib/python3.7/site-packages/conda_build/build.py", line 1505, in bundle_conda
    files = post_process_files(metadata, initial_files)
  File "/Users/runner/miniforge3/lib/python3.7/site-packages/conda_build/build.py", line 1364, in post_process_files
    post_build(m, new_files, build_python=python)
  File "/Users/runner/miniforge3/lib/python3.7/site-packages/conda_build/post.py", line 1178, in post_build
    check_overlinking(m, files, host_prefix)
  File "/Users/runner/miniforge3/lib/python3.7/site-packages/conda_build/post.py", line 1106, in check_overlinking
    m.config.enable_static)
  File "/Users/runner/miniforge3/lib/python3.7/site-packages/conda_build/post.py", line 1068, in check_overlinking_impl
    raise OverLinkingError(overlinking_errors)
conda_build.exceptions.OverLinkingError: overlinking check failed 
['  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/ext_password.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!', '  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/ext_rfc5801.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!', '  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/creds.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!', '  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/names.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!', '  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/types.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!', '  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/ext_set_cred_opt.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!', '  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/sec_contexts.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!', '  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/cython_converters.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!', '  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/mech_krb5.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!', '  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/ext_ggf.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!', '  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/message.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!', '  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/ext_rfc5587.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!', '  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/ext_cred_imp_exp.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!', '  ERROR (python-gssapi,lib/python3.8/site-packages/gssapi/raw/misc.cpython-38-darwin.so): Needed DSO /System/Library/Frameworks/GSS.framework/Versions/A/GSS not found in any CDT/compiler package, nor the whitelist?!']
```
</details>

**Comment by rzillman** [comment]
I am happy to be a maintainer.

**Comment by isuruf** [comment]
For the osx failure, I suggest patching out the lines  https://github.com/pythongssapi/python-gssapi/blob/dac08a83ebe14c2b4919565628f2a7bef4ebbdd7/setup.py#L46-L48

**Comment by dhirschfeld** [comment]
OSX failed. I think the relevant error is:
```
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/tmp/pip-req-build-xggrpc2u/setup.py", line 171, in <module>
        GSSAPI_LIB = ctypes.CDLL(os.path.join(main_path, main_lib))
      File "/Users/runner/miniforge3/conda-bld/python-gssapi_1598487043913/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.8/ctypes/__init__.py", line 373, in __init__
        self._handle = _dlopen(self._name, mode)
    OSError: dlopen(libgssapi_krb5.so, 6): image not found
```
...which is in part of their *"support detection"*:
https://github.com/pythongssapi/python-gssapi/blob/master/setup.py#L174

On Windows I had to short-circuit this detection by providing the `GSSAPI_MAIN_LIB` env var directly.

**Comment by dhirschfeld** [comment]
Poking about in the [OSX package](https://anaconda.org/conda-forge/krb5/1.17.1/download/osx-64/krb5-1.17.1-h75d18d8_2.tar.bz2) I can't find `libgssapi_krb5.so` but there is a `dylib` file of the same name:
```
    Directory: C:\temp\deleteme\lib


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----       18/08/2020   2:31 AM         262972 libgssapi_krb5.2.2.dylib
-a---l       27/08/2020  10:24 AM              0 libgssapi_krb5.2.dylib
-a---l       27/08/2020  10:24 AM              0 libgssapi_krb5.dylib
-a----       18/08/2020   2:31 AM         105956 libgssrpc.4.2.dylib
-a---l       27/08/2020  10:24 AM              0 libgssrpc.4.dylib
-a---l       27/08/2020  10:24 AM              0 libgssrpc.dylib
```

**Comment by dhirschfeld** [comment]
Thanks @isuruf - also patching to look for a `dylib` seemed to do the job!

I think this is good to go...

**Pull request merged**